pkg/apis/version/conversion_test.go

```diff
@@ -29,26 +30,51 @@ type testStruct struct {
 }
 
 func TestSerializationRoundTrip(t *testing.T) {
-	meta := metav1.ObjectMeta{}
-	source := testStruct{Field: "foo"}
-	key := "my-key"
-	err := version.SerializeToMetadata(&meta, source, key)
-	if err != nil {
-		t.Fatalf("Serialization error: %s", err)
-	}
+	var arraySourceType = reflect.TypeOf([]testStruct{{Field: ""}})
 
-	sink := testStruct{}
-	err = version.DeserializeFromMetadata(&meta, &sink, key)
-	if err != nil {
-		t.Fatalf("Deserialization error: %s", err)
-	}
+	testcases := []struct {
+		name   string
+		meta   metav1.ObjectMeta
+		source interface{}
+		key    string
+	}{{
+		name:   "serialization roundtrip",
+		meta:   metav1.ObjectMeta{},
+		source: testStruct{Field: "foo"},
+		key:    "my-key",
+	}, {
+		name:   "array serialization roundtrip",
+		meta:   metav1.ObjectMeta{},
+		source: []testStruct{{Field: "foo"}, {Field: "bar"}},
+		key:    "my-key",
+	}}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := version.SerializeToMetadata(&tc.meta, tc.source, tc.key)
+			if err != nil {
+				t.Fatalf("Serialization error: %s", err)
+			}
+			var sink interface{}
```

Comment:
Wondering if there's any better way to change the variable type here?

Reply:
ah sorry I didn't realize this was why you created two separate test classes. You could put the sink as part of the test case?

Also I think lines 60 and 74 can be removed.

Reply:
It doesn't seem to work fine if I put the test case as 
```
	testcases := []struct {
		name   string
		meta   metav1.ObjectMeta
		sink   interface{}
		source interface{}
		key    string
	}```
The result would be a map instead of the struct.